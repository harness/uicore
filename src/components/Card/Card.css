div.card {
  border-radius: 5px;
  display: inline-block;
  position: relative;
  border: 1px solid transparent;

  &:focus {
    box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.1), 0 2px 4px rgba(16, 22, 26, 0.2), 0 8px 24px rgba(16, 22, 26, 0.2);
    outline: none;
  }
}

.interactive {
  &:hover {
    cursor: pointer;
    box-shadow: 0px 0px 2px rgba(40, 41, 61, 0.04), 0px 4px 8px rgba(96, 97, 112, 0.16) !important;
    border: 1px solid var(--blue-400);
  }
}

.cardIcon {
  :global {
    .bp3-icon {
      display: block;
      svg {
        margin: 0 auto var(--spacing-small);
      }
    }
  }
}

.selected {
  border: 2px solid var(--blue-500) !important;
}

.disabled {
  cursor: not-allowed !important;
  background-color: var(--grey-200) !important;
  color: var(--disabled-color) !important;
}

.cardMenu {
  position: relative;
  &.colorIdentifier {
    &:before,
    &:after {
      content: '';
      position: absolute;
      top: -21px;
      left: -21px;
      border-color: transparent;
      border-style: solid;
    }
    &::after {
      border-top-left-radius: 5px;
      border-width: 15px;
      border-left-color: currentColor;
      border-top-color: currentColor;
    }
  }

  .title {
    margin-left: 5px;
    position: relative;
    top: -10px;
    left: 7px;
    width: calc(100% - 20px);
  }

  .dots {
    position: absolute;
    top: -15px;
    right: -19px;

    .menu {
      cursor: pointer;
      outline: none !important;
      color: var(--grey-500);

      &:hover {
        background-color: var(--grey-200) !important;
        color: var(--blue-500);
      }
    }
  }
}

.corner {
  position: absolute;
  top: 0;
  right: 0;
  .badge {
    width: 0;
    position: absolute;
    height: 0;
    border-top: 20px solid var(--blue-500);
    border-left: 20px solid transparent;
    right: 0;
    z-index: 1;
  }
  :global {
    .bp3-icon {
      position: absolute;
      right: 1px;
      top: 1px;
      z-index: 10;
    }
  }
}
