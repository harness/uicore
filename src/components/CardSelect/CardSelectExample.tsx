/* eslint-disable react/display-name */
import React, { useState } from 'react'
import { CardSelect, Text, CardBody } from '../..'

interface Data {
  text: string
  value: string
  icon: string
}

const data: Data[] = [
  {
    text: 'Kubernetes',
    value: 'service-kubernetes',
    icon: 'service-kubernetes'
  },
  {
    text: 'Github',
    value: 'service-github',
    icon: 'service-github'
  },
  {
    text: 'GCP',
    value: 'service-gcp',
    icon: 'service-gcp'
  },
  {
    text: 'ELK Service',
    value: 'service-elk',
    icon: 'service-elk'
  },
  {
    text: 'Git Labs',
    value: 'service-gotlab',
    icon: 'service-gotlab'
  },
  {
    text: 'Datadog',
    value: 'service-datadog',
    icon: 'service-datadog'
  },
  {
    text: 'Slack',
    value: 'service-slack',
    icon: 'service-slack'
  },
  {
    text: 'Jenkins',
    value: 'service-jenkins',
    icon: 'service-jenkins'
  }
]

export const CardSelectExample = () => {
  const [selected, setSelected] = useState(data[2])
  const props = {
    data,
    className: 'grid',
    renderItem: (item: any, selected: any) => (
      <CardBody.Icon icon={item.icon} iconSize={25}>
        <Text font={{ size: 'small', align: 'center' }} color={selected ? 'var(--grey-900)' : 'var(--grey-350)'}>
          {item.text}
        </Text>
      </CardBody.Icon>
    ),
    cornerSelected: true,
    onChange: (value: any) => setSelected(value)
  }
  return <CardSelect {...props} selected={selected} />
}

export const CardMultiSelectExample = () => {
  const [selected, setSelected] = useState([data[2], data[4]])

  const onChange = React.useCallback(
    value => {
      const selectedAr = [...selected]
      const index = selectedAr.indexOf(value)
      if (index > -1) {
        selectedAr.splice(index, 1)
      } else {
        selectedAr.push(value)
      }
      setSelected(selectedAr)
    },
    [selected]
  )
  const props = {
    data,
    multi: true,
    className: 'grid',
    renderItem: (item: any, selected: any) => (
      <CardBody.Icon icon={item.icon} iconSize={25}>
        <Text font={{ size: 'small', align: 'center' }} color={selected ? 'var(--grey-900)' : 'var(--grey-350)'}>
          {item.text}
        </Text>
      </CardBody.Icon>
    ),
    onChange
  }
  return <CardSelect {...props} selected={selected} />
}
