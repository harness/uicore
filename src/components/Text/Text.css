.lineclamp {
  display: -webkit-box !important;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--text-line-clamp);
  overflow: hidden;
  word-break: break-all;
}

.withDocsTooltip {
  display: flex;
  align-items: center;
}

.targetClass {
  display: flex !important;
}
.icon {
  vertical-align: middle;
}
