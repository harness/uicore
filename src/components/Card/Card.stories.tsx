import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { Layout, Card, Text, Icon, CardProps, Container } from '../..'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary, Description } from '@storybook/addon-docs/blocks'
import { CardBody, Menu, Icon as CardBodyIcon } from './Card'

export default {
  title: 'Components / Card',
  component: Card,
  subcomponents: { ['CardBody.Menu']: Menu, ['CardBody.Icon']: CardBodyIcon },
  parameters: {
    layout: 'centered',
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>Card</Title>
            <Description>{`Card component reuses Blueprint's card and implements the look and feel of [CD Next Gen](https://harness.atlassian.net/wiki/spaces/UI/pages/668271019/Card)`}</Description>
            <Subtitle>
              <>
                <h4>Import</h4>
                <pre>
                  <code>{`import { Card, CardBody }  from '@wings-software/uicore'`}</code>
                </pre>
                <h4>Types</h4>
                <pre>
                  <code>
                    {`interface CardProps extends ICardProps {
  selected?: boolean
  disabled?: boolean
  cornerSelected?: boolean
}
interface ICardProps extends IProps, HTMLDivProps {
    elevation?: Elevation;
    interactive?: boolean;
    onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
}`}
                  </code>
                </pre>
              </>
            </Subtitle>
            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  }
} as Meta

export const Basic: Story<CardProps> = args => (
  <Layout.Horizontal spacing="large">
    <Card interactive {...args}>
      <CardBody.Menu menuContent={<Container padding="medium">hello</Container>} />
      <div style={{ height: '200px', width: '150px' }}>
        <Layout.Vertical spacing="large">
          <div>
            <Icon name="service-jenkins" size={30} />
            <Text style={{ marginTop: '5px' }} font="medium">
              Jenkin Artifact
            </Text>
          </div>
          <div>
            <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
              GETTING THE ARTEFACT
            </Text>
            <Text font="small">Referencing the configuration file</Text>
          </div>
          <div>
            <hr style={{ margin: '0px 0px 10px', border: '1px dashed var(--grey-350)' }} />
            <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
              ARTEFACT SOURCE
            </Text>
            <Text font="small">Fixed</Text>
          </div>
        </Layout.Vertical>
      </div>
    </Card>
    <Card interactive {...args}>
      <div style={{ height: '200px', width: '150px' }}>
        <Layout.Vertical spacing="large">
          <div>
            <Icon name="service-bamboo" size={30} />
            <Text style={{ marginTop: '5px' }} font="medium">
              Bamboo Artifact
            </Text>
          </div>
          <div>
            <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
              GETTING THE ARTEFACT
            </Text>
            <Text font="small">Referencing the configuration file</Text>
          </div>
          <div>
            <hr style={{ margin: '0px 0px 10px', border: '1px dashed var(--grey-350)' }} />
            <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
              ARTEFACT SOURCE
            </Text>
            <Text font="small">Fixed</Text>
          </div>
        </Layout.Vertical>
      </div>
    </Card>
  </Layout.Horizontal>
)
