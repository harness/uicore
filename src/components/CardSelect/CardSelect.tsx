import React, { useRef, useEffect, useCallback } from 'react'
import { Card, CardProps } from '../Card/Card'
import { HTMLDivProps } from '@blueprintjs/core'

export enum CardSelectType {
  CardView = 'CardView',
  Any = 'any'
}

export interface CardSelectProps<ObjectType> extends Omit<HTMLDivProps, 'onChange'> {
  selected: ObjectType | ObjectType[] | undefined
  data: ObjectType[]
  multi?: boolean
  type?: CardSelectType
  cardClassName?: string
  renderItem: (item: ObjectType, selected: boolean) => JSX.Element
  onChange: (selected: ObjectType, e: React.MouseEvent<HTMLDivElement>) => void
}

enum Keys {
  Enter = 13,
  MoveLeft = 37,
  MoveRight = 39
}

const ValidKeys = [Keys.Enter, Keys.MoveLeft, Keys.MoveRight]

export function CardSelect<ObjectType>(props: CardSelectProps<ObjectType>) {
  const {
    className = '',
    cardClassName = '',
    data = [],
    selected,
    renderItem,
    onChange,
    style = {},
    type = CardSelectType.CardView,
    multi = false
  } = props
  const rootRef = useRef<HTMLDivElement>(null)

  const handleUserKeyPress = useCallback(event => {
    const { keyCode, target } = event
    if (ValidKeys.indexOf(keyCode) > -1) {
      if (keyCode === Keys.Enter) {
        target.click()
      } else if (keyCode === Keys.MoveLeft) {
        target.previousSibling?.click()
        target.previousSibling?.focus()
      } else if (keyCode === Keys.MoveRight) {
        target.nextSibling?.click()
        target.nextSibling?.focus()
      }
    }
  }, [])

  useEffect(() => {
    rootRef.current?.addEventListener('keydown', handleUserKeyPress)
    return () => {
      rootRef.current?.removeEventListener('keydown', handleUserKeyPress)
    }
  }, [])

  if (multi && selected && !Array.isArray(selected)) {
    throw 'For CardSelect with multi as true, then selected prop should be an array'
  }
  return (
    <div className={className} style={style} ref={rootRef}>
      {data.length > 0 &&
        data.map((item, index) => {
          const isSelected = multi && Array.isArray(selected) ? selected.indexOf(item) > -1 : item === selected

          return (
            <React.Fragment key={index}>
              {type === CardSelectType.CardView ? (
                <Card
                  className={cardClassName}
                  interactive={true}
                  data-index={index}
                  selected={isSelected}
                  disabled={(item as CardProps).disabled || false}
                  onClick={event => onChange(item, event)}>
                  {renderItem(item, isSelected)}
                </Card>
              ) : (
                <div className={cardClassName} data-index={index} onClick={event => onChange(item, event)}>
                  {renderItem(item, isSelected)}
                </div>
              )}
            </React.Fragment>
          )
        })}
    </div>
  )
}
