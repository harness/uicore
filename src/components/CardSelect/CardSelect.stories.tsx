/* eslint-disable react/display-name */
import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { CardSelect, Text, Layout } from '../..'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary, Description } from '@storybook/addon-docs/blocks'
import { CardSelectProps } from '../CardSelect/CardSelect'
import { CardSelectExample, CardMultiSelectExample } from './CardSelectExample'

export default {
  title: 'Components / CardSelect',

  component: CardSelect,
  parameters: {
    layout: 'centered',
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>CardSelect</Title>

            <Description>{`Design is a like card select component [CD Next Gen](https://harness.atlassian.net/wiki/spaces/UI/pages/668205796/Radio+Button+Select)

Should support

- Dynamic Cards of any design
- Keyboard Events - left arrow, right arrow and Enter

Internally it uses [Card](/card) component and define renderItem method to define the cody of the card and you use class to customize the card
`}</Description>
            <Subtitle>
              <>
                <pre>
                  <code>{`import { CardSelect }  from '@wings-software/uicore'`}</code>
                </pre>
                <pre>
                  <code>{`interface Data {
  text: string
  value: string
  icon: string
}

const data: Data[] = [
  {
    text: 'Kubernetes',
    value: 'service-kubernetes',
    icon: 'service-kubernetes'
  },
  {
    text: 'Github',
    value: 'service-github',
    icon: 'service-github'
  },
  {
    text: 'GCP',
    value: 'service-gcp',
    icon: 'service-gcp'
  },
  {
    text: 'ELK Service',
    value: 'service-elk',
    icon: 'service-elk'
  },
  {
    text: 'Git Labs',
    value: 'service-gotlab',
    icon: 'service-gotlab'
  },
  {
    text: 'Datadog',
    value: 'service-datadog',
    icon: 'service-datadog'
  },
  {
    text: 'Slack',
    value: 'service-slack',
    icon: 'service-slack'
  },
  {
    text: 'Jenkins',
    value: 'service-jenkins',
    icon: 'service-jenkins'
  }
]


export const CardSelectExample = () => {
  const [selected, setSelected] = useState(null)
  const props = {
    data,
    className: 'grid',
    renderItem: (item, selected) => (
      <CardBody.Icon icon={item.icon} iconSize={25}>
        <Text font={{ size: 'small', align: 'center', color: selected ? 'var(--grey-900)' : 'var(--grey-350)' }}>
          {item.text}
        </Text>
      </CardBody.Icon>
    ),
    onChange: value => setSelected(value)
  }
  return <CardSelect {...props} selected={selected} />
}

export const CardMultiSelectExample = () => {
  const [selected, setSelected] = useState([data[2], data[4]])

  const onChange = React.useCallback(
    value => {
      const selectedAr = [...selected]
      const index = selectedAr.indexOf(value)
      if (index > -1) {
        selectedAr.splice(index, 1)
      } else {
        selectedAr.push(value)
      }
      setSelected(selectedAr)
    },
    [selected]
  )
  const props = {
    data,
    multi: true,
    className: 'grid',
    renderItem: (item, selected) => (
      <CardBody.Icon icon={item.icon} iconSize={25}>
        <Text font={{ size: 'small', align: 'center', color: selected ? 'var(--grey-900)' : 'var(--grey-350)' }}>
          {item.text}
        </Text>
      </CardBody.Icon>
    ),
    onChange
  }
  return <CardSelect {...props} selected={selected} />
}`}</code>
                </pre>
              </>
            </Subtitle>
            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  },
  decorators: [Story => <Story />]
} as Meta

export const Select: Story<CardSelectProps<any>> = _args => {
  return (
    <Layout.Vertical spacing="large">
      <Text>Control are not connected to example</Text>
      <CardSelectExample />
    </Layout.Vertical>
  )
}
export const MultiSelect: Story<CardSelectProps<any>> = _args => {
  return (
    <Layout.Vertical spacing="large">
      <Text>Control are not connected to example</Text>
      <CardMultiSelectExample />
    </Layout.Vertical>
  )
}
