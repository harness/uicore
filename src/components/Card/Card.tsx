import React, { useCallback } from 'react'
import { Card as BpCard, ICardProps, HTMLDivProps, Position } from '@blueprintjs/core'
import css from './Card.css'
import cx from 'classnames'
import { Icon as UiIcon, IconName, IconProps } from '../../icons/Icon'
import { Color } from '../../core/Color'
import { Popover, PopoverProps } from '../../components/Popover/Popover'
import { Text } from '../../components/Text/Text'
import { Button } from '../../components/Button/Button'

export interface CardProps extends ICardProps {
  selected?: boolean
  disabled?: boolean
  cornerSelected?: boolean
}

const renderCornerTick = () => {
  return (
    <span className={css.corner}>
      <span className={css.badge}></span>
      <UiIcon name="tick" size={9} color={'white'} />
    </span>
  )
}

export const Card: React.FC<CardProps> = props => {
  const {
    selected = false,
    disabled = false,
    className = '',
    style,
    cornerSelected = false,
    tabIndex = 0,
    ...bpProps
  } = props
  // if card is disabled then interactivity should be false
  if (disabled) {
    bpProps.interactive = false
  }
  const tabIndexProps = bpProps.interactive ? { tabIndex } : {}
  const onClick = useCallback(
    event => {
      if (!event.target.closest(`.${css.dots}`)) {
        props.onClick?.(event)
      }
    },
    [props.onClick]
  )

  return (
    <BpCard
      {...bpProps}
      {...tabIndexProps}
      className={cx(
        css.card,
        className,
        { [css.selected]: selected },
        { [css.disabled]: disabled },
        { [css.interactive]: bpProps.interactive }
      )}
      style={style}
      onClick={onClick}>
      {selected && cornerSelected && renderCornerTick()}
      {props.children}
    </BpCard>
  )
}

// Card Body
interface CardIconProps extends HTMLDivProps {
  icon: IconName
  iconSize?: number
  iconProps?: Partial<IconProps>
}

export const Icon: React.FC<CardIconProps> = props => {
  const { icon, iconSize = 16, iconProps = {}, className, style } = props

  return (
    <div className={cx(css.cardIcon, className)} style={style}>
      <UiIcon name={icon} size={iconSize} {...iconProps} />
      {props.children}
    </div>
  )
}

interface CardMenuProps extends HTMLDivProps {
  menuContent: JSX.Element
  title?: string
  colorIdentifier?: Color
  menuPopoverProps?: PopoverProps
}

export const Menu: React.FC<CardMenuProps> = props => {
  const { menuContent, title, colorIdentifier = false, className, style = {}, menuPopoverProps } = props
  // NextJS does not work well with usePortal={true}
  const isNext =
    typeof window !== 'undefined' && typeof window.next !== 'undefined' && typeof window.__NEXT_DATA__ !== 'undefined'

  return (
    <div
      className={cx(css.cardMenu, className, { [css.colorIdentifier]: colorIdentifier })}
      style={{ color: colorIdentifier ? colorIdentifier : '', ...style }}>
      <div className={css.dots}>
        <Popover content={menuContent} position={Position.RIGHT_TOP} usePortal={!isNext} {...menuPopoverProps}>
          <Button className={css.menu} minimal round icon="Options" />
        </Popover>
      </div>
      {title && (
        <Text className={css.title} font="medium">
          {title}
        </Text>
      )}
      {props.children}
    </div>
  )
}

export const CardBody = { Icon, Menu }
