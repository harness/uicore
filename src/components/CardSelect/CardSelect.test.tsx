/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { CardSelect } from './CardSelect'
import { CardBody } from '../Card/Card'
import { Text } from '../Text/Text'
import { IconName } from 'icons/Icon'

interface Data {
  text: string
  value: string
  icon: IconName
  disabled?: boolean
}

const data: Data[] = [
  {
    text: 'Kubernetes',
    value: 'service-kubernetes',
    icon: 'service-kubernetes'
  },
  {
    text: 'Github',
    value: 'service-github',
    icon: 'service-github'
  },
  {
    text: 'GCP',
    value: 'service-gcp',
    icon: 'service-gcp'
  },
  {
    text: 'ELK Service',
    value: 'service-elk',
    icon: 'service-elk'
  },
  {
    text: 'Git Labs',
    value: 'service-gotlab',
    icon: 'service-gotlab'
  },
  {
    text: 'Datadog',
    value: 'service-datadog',
    icon: 'service-datadog'
  },
  {
    text: 'Slack',
    value: 'service-slack',
    icon: 'service-slack'
  },
  {
    text: 'Jenkins',
    value: 'service-jenkins',
    icon: 'service-jenkins'
  }
]

const getDefaultProps = (item?: Data) => ({
  data,
  className: 'grid',
  // eslint-disable-next-line react/display-name
  renderItem: (item: Data, selected: boolean) => (
    <CardBody.Icon icon={item.icon} iconSize={25}>
      <Text color={selected ? 'var(--grey-900)' : 'var(--grey-350)'}>{item.text}</Text>
    </CardBody.Icon>
  ),
  onChange: jest.fn(),
  selected: item
})

describe('Test render Radio Select', () => {
  test('should render default values', () => {
    const { container } = render(<CardSelect {...getDefaultProps()} />)
    expect(container).toMatchSnapshot()
  })
  test('should render with selected values', () => {
    const { container } = render(<CardSelect {...getDefaultProps(data[2])} />)
    expect(container).toMatchSnapshot()
  })
  test('should render with disabled value', () => {
    const props = getDefaultProps()
    props.data[2] = { ...props.data[2], disabled: true }
    const { container } = render(<CardSelect {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('should handle on change Event', async () => {
    const props = getDefaultProps(data[2])
    const selectedIndex = 5
    const { container } = render(<CardSelect {...props} />)
    fireEvent.click(container.querySelectorAll('.bp3-card')[selectedIndex])
    await wait()
    expect(props.onChange).toHaveBeenCalledWith(data[selectedIndex], expect.any(Object))
  })
  test('should handle on arrow key Event', async () => {
    const defaultIndex = 2
    const props = getDefaultProps(data[defaultIndex])
    const { container } = render(<CardSelect {...props} />)

    // Left Arrow
    fireEvent.keyDown(container.querySelectorAll('.bp3-card')[defaultIndex], {
      key: 'ArrowLeft',
      keyCode: 37,
      which: 37
    })
    await wait()
    expect(props.onChange).toHaveBeenCalledWith(data[defaultIndex - 1], expect.any(Object))

    // Right Arrow
    fireEvent.keyDown(container.querySelectorAll('.bp3-card')[defaultIndex], {
      key: 'ArrowRight',
      keyCode: 39,
      which: 39
    })
    await wait()
    expect(props.onChange).toHaveBeenCalledWith(data[defaultIndex + 1], expect.any(Object))
  })

  test('should handle on Enter key Event', async () => {
    const defaultIndex = 2
    const selectedIndex = 5
    const props = getDefaultProps(data[defaultIndex])
    const { container } = render(<CardSelect {...props} />)

    // Enter Key
    fireEvent.keyDown(container.querySelectorAll('.bp3-card')[selectedIndex], {
      key: 'Enter',
      keyCode: 13,
      which: 14
    })
    await wait()
    expect(props.onChange).toHaveBeenCalledWith(data[selectedIndex], expect.any(Object))
  })
})
