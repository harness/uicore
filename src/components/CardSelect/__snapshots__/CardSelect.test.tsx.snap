// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Radio Select should render default values 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="0"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-kubernetes"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Kubernetes
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="1"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-github"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Github
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="2"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gcp"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          GCP
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="3"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-elk"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          ELK Service
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="4"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gotlab"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Git Labs
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="5"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-datadog"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Datadog
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="6"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-slack"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Slack
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="7"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-jenkins"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Jenkins
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test render Radio Select should render with disabled value 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="0"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-kubernetes"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Kubernetes
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="1"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-github"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Github
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-elevation-0 card disabled"
      data-index="2"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gcp"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          GCP
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="3"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-elk"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          ELK Service
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="4"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gotlab"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Git Labs
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="5"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-datadog"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Datadog
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="6"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-slack"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Slack
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="7"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-jenkins"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Jenkins
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test render Radio Select should render with selected values 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="0"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-kubernetes"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Kubernetes
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="1"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-github"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Github
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card selected"
      data-index="2"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gcp"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-900)"
        >
          GCP
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="3"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-elk"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          ELK Service
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="4"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-gotlab"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Git Labs
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="5"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-datadog"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Datadog
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="6"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-slack"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Slack
        </p>
      </div>
    </div>
    <div
      class="bp3-card bp3-interactive bp3-elevation-0 card"
      data-index="7"
      tabindex="0"
    >
      <div
        class="cardIcon"
      >
        <span
          class="bp3-icon main"
          data-icon="service-jenkins"
        >
          <svg
            height="25"
            width="25"
          />
        </span>
        <p
          class="font main color color-var(--grey-350)"
        >
          Jenkins
        </p>
      </div>
    </div>
  </div>
</div>
`;
